src/sagemaker_containers/serializers.py

```diff
@@ -0,0 +1,185 @@
+from six import BytesIO, StringIO
+
+from sagemaker_containers import content_types
+
+
+class Npy(object):
+    """Npy Serializer"""
+
+    @staticmethod
+    def loads(obj):  # type: (object) -> object
+        """Load arrays or pickled objects from memory into numpy arrays.
+
+        Args:
+            obj (object): object to be deserialized.
+
+        Returns:
+            (np.array): deserialized numpy array.
+        """
+        stream = BytesIO(obj)
+        return np.load(stream)
+
+    @staticmethod
+    def dumps(nparray):  # type: (np.array) -> object
+        """Serializes an numpy array into memory.
+
+        Args:
+            nparray (np.array): np array object to be serialized.
+
+        Returns:
+            (obj): serialized np array.
+        """
+        buffer = BytesIO()
+        np.save(buffer, nparray)
+        return buffer.getvalue()
+
+
+class Json(object):
+    """JSON serializer"""
+
+    @staticmethod
+    def loads(string):  # type: (str) -> object
+        """Deserialize a JSON string in a Python object.
+
+        Args:
+            string (str): JSON string to be deserialized
+
+        Returns:
+            (object): deserialized Python object.
+        """
+        return stdjson.loads(string)
+
+    @staticmethod
+    def _numpy_dumps(obj):
+        try:
+            return stdjson.dumps(obj.tolist())
+        except (TypeError, SyntaxError, AttributeError):
+            return None
+
+    @staticmethod
+    def _stream_dumps(obj):
+        try:
+            return stdjson.dumps(obj.read())
+        except (TypeError, SyntaxError, AttributeError):
+            return None
+
+    def dumps(self, obj):  # type: (object) -> str
+        """Serializes python objects, streams, numpy arrays into JSON.
+
+        Args:
+            obj (object): object to be serialized.
+
+        Returns:
+            str:  serialized object
+        """
+        return self._numpy_dumps(obj) or self._stream_dumps(obj) or stdjson.dumps(obj)
+
+
+class Csv(object):
+    """JSON serializer"""
+
+    @staticmethod
+    def loads(string):  # type: (str) -> object
+        """Deserialize a CSV string in a Python object.
+
+        Args:
+            string (str): CSV string to be deserialized
+
+        Returns:
+            (object): deserialized Python object.
+        """
+        try:
+            string = string.decode('utf-8')
+        except AttributeError:
+            pass
+        stream = StringIO(string)
+        return np.genfromtxt(stream, dtype=np.float32, delimiter=',')
+
+    @staticmethod
+    def dumps(obj):  # type: (object) -> str
+        """Serializes python objects, streams, numpy arrays into CSV.
+
+        Args:
+            obj (object): object to be serialized.
+
+        Returns:
+            str:  deserialized object.
+        """
+        stream = StringIO()
+        np.savetxt(stream, obj, delimiter=',', fmt='%s')
+        return stream.getvalue()
+
+
+def dumps(obj, content_type):
```

Comment:
top level... why? put them in a DefaultSerializer class.

Reply:
Classes why? Look how python does: https://github.com/python/cpython/blob/master/Lib/json/__init__.py#L183